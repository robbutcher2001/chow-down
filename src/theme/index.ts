import { DefaultTheme } from 'styled-components';

import colour from './defs/_colour';

const theme: DefaultTheme = {
  colour
};

export default theme;