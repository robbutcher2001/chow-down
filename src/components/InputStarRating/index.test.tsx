import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '../../themes';
import StarRating from '.';

const mockFormState = {
  rating: 2
};

test('StarRating basic snapshot render', () => {
  const starRating = renderer.create(
    <ThemeProvider theme={theme}>
      <StarRating
        name='title'
        label='The Star Rating Title'
        form={mockFormState}
        setNewFormState={jest.fn}
      />
    </ThemeProvider>
  );

  expect(starRating.toJSON()).toMatchSnapshot();
});

test('StarRating correct tag content assertion', () => {
  const { getByText } = render(
    <ThemeProvider theme={theme}>
      <StarRating
        name='title'
        label='The Star Rating Title'
        form={mockFormState}
        setNewFormState={jest.fn}
      />
    </ThemeProvider>
  );

  expect(getByText(/The Star Rating Title/).textContent).toEqual('The Star Rating Title');
});