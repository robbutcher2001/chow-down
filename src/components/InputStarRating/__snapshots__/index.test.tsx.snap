// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating basic snapshot render 1`] = `
<label
  className="sc-bdVaJa egHsBw"
  htmlFor="title"
>
  The Star Rating Title
  <div
    id="title"
  >
    <label
      className="empty"
      htmlFor="rating_1"
    >
      <input
        hidden={true}
        id="rating_1"
        name="title"
        onChange={[Function]}
        type="radio"
        value="1"
      />
    </label>
    <label
      className="empty"
      htmlFor="rating_2"
    >
      <input
        hidden={true}
        id="rating_2"
        name="title"
        onChange={[Function]}
        type="radio"
        value="2"
      />
    </label>
    <label
      className="empty"
      htmlFor="rating_3"
    >
      <input
        hidden={true}
        id="rating_3"
        name="title"
        onChange={[Function]}
        type="radio"
        value="3"
      />
    </label>
    <label
      className="empty"
      htmlFor="rating_4"
    >
      <input
        hidden={true}
        id="rating_4"
        name="title"
        onChange={[Function]}
        type="radio"
        value="4"
      />
    </label>
    <label
      className="empty"
      htmlFor="rating_5"
    >
      <input
        hidden={true}
        id="rating_5"
        name="title"
        onChange={[Function]}
        type="radio"
        value="5"
      />
    </label>
  </div>
</label>
`;
